File: randomizer/Graph/ScriptTemplate/Round/CGRound.lua
local CGRound= CGRound or {}
CGRound.__index = CGRound

function CGRound.new()
    local self = setmetatable({}, CGRound)
    self.inputs = {}
    return self
end

function CGRound:setInput(index, func)
    self.inputs[index] = func
end

function CGRound:getOutput(index)
    return math.floor(self.inputs[0]() + 0.5)
end

return CGRound

File: randomizer/Graph/ScriptTemplate/Round/CGRound.js
/**
 * @file CGRound.js
 * @author liujiacheng
 * @date 2021/8/23
 * @brief CGRound.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGRound extends BaseNode {
  constructor() {
    super();
  }

  setNext(index, func) {
    this.nexts[index] = func;
  }

  setInput(index, func) {
    this.inputs[index] = func;
  }

  getOutput() {
    return Math.floor(this.inputs[0]() + 0.5);
  }
}

exports.CGRound = CGRound;
